addons/event/models/res_partner.py

```diff
@@ -20,3 +24,22 @@ def action_event_view(self):
         action['context'] = {}
         action['domain'] = [('registration_ids.partner_id', 'child_of', self.ids)]
         return action
+
+    def google_map_signed_img(self, zoom=8, width=298, height=298):
+        """Create a signed static image URL for the location of this partner."""
+        GOOGLE_MAPS_STATIC_API_KEY = self.env['ir.config_parameter'].sudo().get_param('event.google_maps_static_api_key')
+        GOOGLE_MAPS_STATIC_API_SECRET = self.env['ir.config_parameter'].sudo().get_param('event.google_maps_static_api_secret')
+        if not (GOOGLE_MAPS_STATIC_API_KEY and GOOGLE_MAPS_STATIC_API_SECRET):
+            return False
+        # generate signature as per https://developers.google.com/maps/documentation/maps-static/digital-signature#server-side-signing
+        params = {
+            'center': '%s, %s %s, %s' % (self.street or '', self.city or '', self.zip or '', self.country_id and self.country_id.display_name or ''),
+            'size': "%sx%s" % (width, height),
+            'zoom': zoom,
+            'sensor': 'false',
+            'key': GOOGLE_MAPS_STATIC_API_KEY,
+        }
+        if GOOGLE_MAPS_STATIC_API_SECRET:
+            unsigned_path = '/maps/api/staticmap?' + werkzeug.urls.url_encode(params)
+            params['signature'] = base64.urlsafe_b64encode(hmac.digest(base64.urlsafe_b64decode(GOOGLE_MAPS_STATIC_API_SECRET), bytes(unsigned_path, 'ascii'), 'sha1'))
+        return 'https://maps.googleapis.com/maps/api/staticmap?' + werkzeug.urls.url_encode(params)
```

Comment:
This whole thing could be moved to the mail module? I'm not sure how relevant it is, maybe it could be useful elsewhere. It would avoid having to get a key per app, but on the other hand it would move the settings for the map to... general settings I guess? 

Not sure.